app/controllers/messages_controller.rb

```diff
@@ -9,8 +9,7 @@ class MessagesController < ApplicationController
   authorize_resource
 
   before_action :lookup_user, :only => [:new, :create]
-  before_action :check_database_readable
-  before_action :check_database_writable, :only => [:new, :create, :reply, :mark, :destroy]
+  before_action :check_api_writable, :only => [:new, :create, :reply, :mark, :destroy]
```

Comment:
Why are we changing the non-API access to check the API status instead of the database status? and why are we getting rid of the readable check for methods like `inbox` and `show`?

Reply:
Thanks for noticing this. I am not sure how it sneaked in.